src/main/java/cash/z/ecc/android/sdk/tool/SaplingParamTool.kt

```diff
@@ -0,0 +1,135 @@
+                ZcashSdk.SPEND_PARAM_FILE_NAME,
+                ZcashSdk.OUTPUT_PARAM_FILE_NAME
+            ).forEach { paramFileName ->
+                val url = "${ZcashSdk.CLOUD_PARAM_DIR_URL}/$paramFileName"
+                val request = Request.Builder().url(url).build()
+                val response = client.newCall(request).execute()
+                if (response.isSuccessful) {
+                    twig("fetch succeeded")
+                    val file = File(destinationDir, paramFileName)
+                    if (file.parentFile.exists()) {
+                        twig("directory exists!")
+                    } else {
+                        twig("directory did not exist attempting to make it")
+                        file.parentFile.mkdirs()
+                    }
+                    Okio.buffer(Okio.sink(file)).use {
+                        twig("writing to $file")
+                        it.writeAll(response.body().source())
+                    }
+                    twig("fetch succeeded, done writing $paramFileName")
+                } else {
+                    failureMessage += "Error while fetching $paramFileName : $response\n"
+                    twig(failureMessage)
+                }
+            }
+            if (failureMessage.isNotEmpty()) throw TransactionEncoderException.FetchParamsException(
+                failureMessage
+            )
+        }
+
+
+        fun clear(destinationDir: String){
+            if(validate(destinationDir)){
+                arrayOf(
+                    ZcashSdk.SPEND_PARAM_FILE_NAME,
+                    ZcashSdk.OUTPUT_PARAM_FILE_NAME
+                ).forEach { paramFileName ->
+                    val file = File(destinationDir, paramFileName)
+                    if(file.deleteRecursively()){
+                        twig("Files deleted successfully")
+                    }else{
+                        twig("Error: Files not able to be deleted!")
+                    }
+                }
+            }
+        }
+
+        fun validate(destinationDir: String): Boolean{
+            arrayOf(
+                ZcashSdk.SPEND_PARAM_FILE_NAME,
+                ZcashSdk.OUTPUT_PARAM_FILE_NAME
+            ).forEach { paramFileName ->
+                val file = File(destinationDir, paramFileName)
+                if(file.exists()){
+                    twig("Files does  exist!")
+                    return true
+                }
+            }
+            twig("Error: File does not exist!")
+            return false
+        }
+
+        //
+        // Helpers
+        //
+        /**
+         * Http client is only used for downloading sapling spend and output params data, which are
+         * necessary for the wallet to scan blocks.
+         *
+         * @return an http client suitable for downloading params data.
+         */
+        private fun createHttpClient(): OkHttpClient {
+            //TODO: add logging and timeouts
+            return OkHttpClient()
```

Comment:
Does this enforce HTTS by default? It's 2021 so I assume it does.

Reply:
I believe it will fetch any URL that it is provided. We could configure it to be more restrictive or just make sure this tool validates what's downloaded. Librustzcash already verifies the checksum of the params before using them. The URL of the params is controlled by the SDK. So I think we're probably okay to leave this as is.